### ISSUE TITLE
Add a prometheus pushgateway input module

### ISSUE THREAD

**Issue by IxDay** [influxdata/telegraf#6418]
I am using telegraf on a box containing a bunch of services using prometheus metrics. I think it could be a good idea to aggregate all those metrics through telegraf before exposing them as a single http endpoint using the output prometheus.

Basically, I will use [the push API from my services](https://godoc.org/github.com/prometheus/client_golang/prometheus/push) to target the telegraf pushgateway endpoint. This will avoid doing complex stuff as files or named pipes.

Looking at [this comment](https://github.com/influxdata/telegraf/issues/4414#issuecomment-530144238) it seems that I am not the only one trying to achieve this.

Thoughts?

**Comment by danielnelson** [comment]
This is a good idea, but there are a few things in flight for prometheus that need to be coordinated.  I need to finish #3670 so that we can merge #5767, at that point we can create the Prometheus parser as mentioned in #4414 (perhaps we would need two parsers, one for text and one for protobuf).

After all this it should be fairly easy to add a pushgateway input.

**Comment by lee3164** [comment]
What's going on with this issue?

**Comment by danielnelson** [comment]
I think the next step is to extract the parser from the prometheus input into a standalone Parser.

**Comment by kushwiz** [comment]
Any updates here?

**Comment by pgmillon** [comment]
I think this has been handled with https://github.com/influxdata/telegraf/pull/7778

**Comment by Catbuttes** [comment]
> I think this has been handled with #7778

I don't believe it has been handled. Attempting to use Telegraf as a replacement for Pushgateway with anything other than extremely simple metric definitions fails with 404 errors due to the way Pushgateway handles labels.

**Comment by srebhan** [comment]
Is anyone still interested in this feature?
